app/models/concerns/foreman_ansible/hostgroup_extensions.rb

```diff
@@ -38,6 +41,16 @@ def all_ansible_roles
         (inherited_ansible_roles + ansible_roles + host_ansible_roles).uniq
       end
     end
+
+    class_methods do
+      def search_by_role(_key, operator, value)
+        conditions = sanitize_sql_for_conditions(["ansible_roles.name #{operator} ?", value_to_sql(operator, value)])
+        hostgroup_ids = ::Hostgroup.unscoped.with_taxonomy_scope.joins(:ansible_roles).where(conditions).map(&:subtree_ids).flatten
```

Comment:
`unscoped` [0] and `with_taxonomy_scope` can be skipped here, because host groups are scoped in taxonomy by default [1].

With this code we reset scope by `unscoped` and set it back with `with_taxonomy_scope`, that just doesn't make sense IMHO.

[0] https://apidock.com/rails/v6.1.3.1/ActiveRecord/Scoping/Default/ClassMethods/unscoped
[1] https://github.com/theforeman/foreman/blob/develop/app/models/hostgroup.rb#L52